### ISSUE TITLE
package.json main field ignored for modules inside asar packaged app

### ISSUE THREAD

**Issue by adill** [electron/electron#18182]
### Preflight Checklist
* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 6.0.0-beta.1
* **Operating System:**
  * Windows 10 Pro (1809)
* **Last Known Working Electron version:**:
  * 5.0.1

### Expected Behavior
`rimraf` should be able to be `require`'d within an ASAR packaged app.

### Actual Behavior
Modules that use `package.json`'s `main` field will fail to load from within the context of an ASAR packaged app. It appears, to me, that `internalModuleReadJSON` from within `readPackage` in the commonjs loader `lib/internal/modules/cjs/loader.js` isn't referencing ASAR-aware `fs` methods.

### To Reproduce
```sh
$ git clone https://github.com/adill/electron-quick-start -b package-json-in-asar
$ cd app\
$ npm install
$ cd ..
$ npm install
$ npm run-script build
$ npm run-script start
```

### Screenshots
![image](https://user-images.githubusercontent.com/144914/57243854-f7fc1000-6feb-11e9-9df3-384e5b1f87f2.png)

**Comment by MarshallOfSound** [comment]
This was fixed in 5.0 and I guess not master / 6-0-x

**Comment by MarshallOfSound** [comment]
This PR needs to land on master and 6-0-x

https://github.com/electron/electron/pull/17906
